posthog/cdp/templates/loops/test_template_loops.py

```diff
@@ -1,61 +1,180 @@
+        assert self.get_mock_print_calls() == snapshot([("No email set. Skipping...",)])
+
+
+class TestTemplateLoopsEvent(BaseHogFunctionTemplateTest):
+    template = template_loops_send_event
+
+    def _inputs(self, **kwargs):
+        inputs = {
+            "apiKey": "1cac089e00a708680bdb1ed9f082d5bf",
+            "email": "max@posthog.com",
+            "include_all_properties": False,
+            "properties": {"product": "PostHog"},
+        }
+        inputs.update(kwargs)
+        return inputs
+
+    def test_function_works(self):
+        self.run_function(
+            inputs=self._inputs(),
+            globals={
+                "person": {
+                    "id": "c44562aa-c649-426a-a9d4-093fef0c2a4a",
+                    "properties": {"name": "Max", "company": "PostHog"},
+                },
+                "event": {
+                    "event": "pageview",
+                    "properties": {"pathname": "/pricing"},
+                },
+            },
+        )
+
+        assert self.get_mock_fetch_calls()[0] == snapshot(
+            (
+                "https://app.loops.so/api/v1/events/send",
+                {
+                    "method": "POST",
+                    "headers": {
+                        "Content-Type": "application/json",
+                        "Authorization": "Bearer 1cac089e00a708680bdb1ed9f082d5bf",
+                    },
+                    "body": {
+                        "email": "max@posthog.com",
+                        "userId": "c44562aa-c649-426a-a9d4-093fef0c2a4a",
+                        "eventName": "pageview",
+                        "product": "PostHog",
+                    },
                 },
             )
+        )
+
+    def test_include_all_properties(self):
+        self.run_function(
+            inputs=self._inputs(include_all_properties=True),
+            globals={
+                "person": {
+                    "id": "c44562aa-c649-426a-a9d4-093fef0c2a4a",
+                    "properties": {"company": "PostHog"},
+                },
+                "event": {
+                    "event": "pageview",
+                    "properties": {"pathname": "/pricing"},
+                },
+            },
+        )
+
+        assert self.get_mock_fetch_calls()[0] == snapshot(
+            (
+                "https://app.loops.so/api/v1/events/send",
+                {
+                    "method": "POST",
+                    "headers": {
+                        "Content-Type": "application/json",
+                        "Authorization": "Bearer 1cac089e00a708680bdb1ed9f082d5bf",
+                    },
+                    "body": {
+                        "email": "max@posthog.com",
+                        "userId": "c44562aa-c649-426a-a9d4-093fef0c2a4a",
+                        "eventName": "pageview",
+                        "product": "PostHog",
+                        "pathname": "/pricing",
+                    },
+                },
```

Comment:
This is almost perfect. Properties related to the event (i.e. from `event.properties`) [should be added to an `eventProperties` object](https://loops.so/docs/api-reference/send-event) instead of in the request root. 

```suggestion
{
                    "method": "POST",
                    "headers": {
                        "Content-Type": "application/json",
                        "Authorization": "Bearer 1cac089e00a708680bdb1ed9f082d5bf",
                    },
                    "body": {
                        "email": "max@posthog.com",
                        "userId": "c44562aa-c649-426a-a9d4-093fef0c2a4a",
                        "eventName": "pageview",
                        "product": "PostHog",
                        "company": "PostHog",
                        "eventProperties": {
                            "pathname": "/pricing",
                        }
                    },
                },
```

Contact/user properties from `person.properties` (i.e. `company`) can be added to the root. Maybe for this you could add another flag `include_all_user_properties` to make this a separate option.

(I'm unsure if the `product` data here in this example is tied to the contact or the event. If it's an event-related piece of data rather than the user/contact, it should be moved to `eventProperties` too.)

Reply:
I've updated the event destination to include all properties in the `eventProperties` object. Contact properties can be updated with the contacts destination.